font-codegen/src/parsing.rs

```diff
@@ -1184,22 +1209,85 @@ impl Item {
     }
 }
 
-impl Parse for Count {
+impl Parse for CountArg {
     fn parse(input: ParseStream) -> syn::Result<Self> {
-        let fork = input.fork();
-        if fork.parse::<Token![$]>().is_ok()
-            && fork.parse::<syn::Ident>().is_ok()
-            && fork.is_empty()
-        {
+        if input.peek(Token![$]) {
             input.parse::<Token![$]>()?;
-            return Ok(Self::Field(input.parse()?));
+            input.parse().map(Self::Field)
+        } else if input.peek(Token![..]) {
+            input.parse().map(Self::All)
+        } else {
+            let int = input.parse::<syn::LitInt>()?;
+            let digits = int.base10_digits();
+            if digits.starts_with('-') {
+                return Err(syn::Error::new(
+                    input.span(),
+                    "negative count is not supported",
+                ));
+            }
+            //HACK: we ensure these literals always have an explicit type.
+            //Is this necessary? no clue.
+            if int.suffix() != "usize" {
+                let reparse = format!("{digits}_usize");
+                syn::parse_str(&reparse).map(Self::Literal)
+            } else {
+                Ok(Self::Literal(int))
+            }
         }
+    }
+}
 
-        if input.peek(Token![..]) {
-            input.parse::<Token![..]>()?;
-            Ok(Self::All)
+impl Parse for Count {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        if input.peek(syn::Ident) {
+            // leading ident must be a function
+            let xform = input.parse()?;
+            let content;
+            let _ = parenthesized!(content in input);
+            let args = Punctuated::<CountArg, Token![,]>::parse_terminated(&content)?
+                .into_iter()
+                .collect();
+            Count::try_from_fancy_stuff(input.span(), xform, args)
         } else {
-            input.parse().map(Self::Expr)
+            input.parse().map(Self::Simple)
+        }
+    }
+}
+
+impl Parse for CountTransform {
+    fn parse(input: ParseStream) -> syn::Result<Self> {
+        let ident = input.parse::<syn::Ident>()?;
+        if ident == "subtract" {
+            Ok(CountTransform::Sub)
+        } else if ident == "half" {
+            Ok(CountTransform::Half)
+        } else if ident == "delta_value_count" {
+            Ok(CountTransform::DeltaValueCount)
+        } else if ident == "map_data_size" {
+            Ok(CountTransform::DeltaSetIndexData)
+        } else {
+            Err(syn::Error::new(ident.span(), "unknown named transform, expected one of 'subtract', 'half' 'delta_value_count', or 'map_data_size'"))
```

Comment:
Can one generate the valid values list? - once you've got 10 of these lists at least one will be wrong